db/db.go

```diff
@@ -2939,37 +2953,27 @@ func GetBLSChangesCountForQuery(query string) (uint64, error) {
 		FROM blocks_bls_change bls
 		INNER JOIN blocks b ON bls.block_root = b.blockroot AND b.status = '1'
 		%s
-		%s`
+		`
 
 	trimmedQuery := strings.ToLower(strings.TrimPrefix(query, "0x"))
 	var err error = nil
 
-	joinQuery := `
-		LEFT JOIN (
-			SELECT 
-				validators.validatorindex as validatorindex,
-				eth1_deposits.from_address_text as deposit_address_text
-			FROM validators 
-			INNER JOIN eth1_deposits ON validators.pubkey = eth1_deposits.publickey
-		) AS val ON val.validatorindex = bls.validatorindex`
-
 	// Check whether the query can be used for a validator, slot or epoch search
 	if uiQuery, parseErr := strconv.ParseUint(query, 10, 64); parseErr == nil {
 		searchQuery := `
 			WHERE bls.validatorindex = $1			
 				OR block_slot = $1
-				OR (block_slot / $3) = $1
-				OR pubkey_text LIKE ($2 || '%')
-				OR deposit_address_text LIKE ($2 || '%')`
```

Comment:
**question:** Why was the address search removed here? Are there not valid addresses that can be parsed as **uint64**?

Reply:
has not been removed, but we only search for exact matches